dvc/command/version.py

```diff
@@ -1,30 +1,107 @@
 from __future__ import unicode_literals
 
+import os
+import psutil
 import platform
 import argparse
 import logging
+import uuid
 
+from dvc.repo import Repo
 from dvc.command.base import CmdBaseNoRepo, append_doc_link
 from dvc.version import __version__
+from dvc.exceptions import DvcException, NotDvcRepoError
+from dvc.system import System
 
 
 logger = logging.getLogger(__name__)
 
 
 class CmdVersion(CmdBaseNoRepo):
     def run(self):
+        dvc_version = __version__
+        python_version = platform.python_version()
+        platform_type = platform.platform()
+
         info = (
             "DVC version: {dvc_version}\n"
             "Python version: {python_version}\n"
-            "Platform: {platform_type}"
+            "Platform: {platform_type}\n"
         ).format(
-            dvc_version=__version__,
-            python_version=platform.python_version(),
-            platform_type=platform.platform(),
+            dvc_version=dvc_version,
+            python_version=python_version,
+            platform_type=platform_type,
         )
+
+        try:
+            root_directory = Repo.find_root()
+
+            info += (
+                "Filesystem type: {filesystem_type}\n" "Cache: {cache}"
+            ).format(
+                filesystem_type=self.get_fs_type(
+                    os.path.abspath(root_directory)
+                ),
+                cache=self.get_linktype_support_info(),
+            )
+
+        except NotDvcRepoError:
+            root_directory = os.getcwd()
+
+            info += ("Filesystem type: {filesystem_type}").format(
+                filesystem_type=self.get_fs_type(
+                    os.path.abspath(root_directory)
+                )
+            )
+
         logger.info(info)
         return 0
 
+    def get_fs_type(self, path):
+        partition = {}
+        for part in psutil.disk_partitions():
+            partition[part.mountpoint] = (part.fstype, part.device)
+        if path in partition:
+            return partition[path]
+        splitpath = path.split(os.sep)
+        for i in range(len(splitpath), 0, -1):
+            path = os.sep.join(splitpath[:i]) + os.sep
+            if path in partition:
+                return partition[path]
+            path = os.sep.join(splitpath[:i])
+            if path in partition:
+                return partition[path]
+        return ("unkown", "none")
+
+    def get_linktype_support_info(self):
+        links = {
+            "reflink": System.reflink,
+            "hardlink": System.hardlink,
+            "symlink": System.symlink,
+        }
+
+        repo = Repo()
+        fname = "." + str(uuid.uuid4())
+        src = os.path.join(repo.cache.local.cache_dir, fname)
```

Comment:
This file doesn't exist :slightly_smiling_face: You need to create it first, then use and then not forget to remove it.

Reply:
@efiop then it should throw a `FileError` when running on local right? Why doesn't it?

Reply:
@efiop
This snippet somehow doesn't create a file in the root directory. 
```
dst = os.path.join(repo.root_dir, fname)
test_file = open(dst, 'w')
```

This creates a file in the cache directory.
```
src = os.path.join(repo.cache.local.cache_dir, fname)
cache_file = open(src, 'w')
```

I tried giving `766` permission to root directory but it doesn't work. What could be the problem?

Reply:
@algomaster99 does it throw an error or something?

Reply:
@efiop It doesn't. It just won't create the file.

Reply:
@algomaster99 and how about
```
dst = os.path.join(repo.root_dir, fname)
test_file = open(dst, 'w').close()
```
?

Reply:
@efiop No, it doesn't work. It is unable to create a file in the root directory.

Reply:
It works for me. Are you sure you are looking at the right place? Also, note that those files are hidden (since the name starts with `.`) so you'll need to do something like `ls -la` to see them.

Reply:
@efiop I am looking at the same level as `.dvc`. The file is being created in the cache directory. I could see it using `ls -al`. I will see if I can figure it out if it works for you.

Reply:
@efiop I tried it on a different system. Doesn't work for me. Is it creating a file in your root directory as well as cache directory?

Reply:
@algomaster99 Wait, so why are you trying to create a file in the repo root? You should create it in cache_dir and then try creating links to it in the repo root.

Reply:
Btw, could you post a script that you are reproducing that issue with? :slightly_smiling_face: